examples/nginx-connect-proxy.md

```diff
@@ -0,0 +1,254 @@
+# NGINX as a Consul Connect ingress proxy
+
+You can use NGINX to proxy into a Consul Connect service mesh without a sidecar
+by fetching the required certificates using Consul-Template. This guide is
+based off the introductory [service mesh learn article][mesh] and, like it,
+reviews a simple example you can run locally. Please see the [mesh][mesh]
+introductory article for more details as this guide will just run through the
+basics to get the example running.
+
+### Security/Techincal Note
+
+To follow this guide you'll need nginx, consul and consul-template installed.
+Additionally the shell script requires some standard Unix CLI tools as well as
+GNU Screen.
+
+This guide is only meant to demonstrate features and many things have been
+simplified. It is not a production ready/secure way to deploy things.
+
+Each of the services below will run in the foreground and output its logs to
+that console. They are designed to all run from files in the same directory.
+
+For convenience sake you can find a shell script to extract the configuration
+files and run the commands the from this document in the
+examples/nginx-connect-proxy/ sub-directory. The script should be a handy
+reference even if you don't want to run it directly.
+
+## Start Consul Agent with the Services Configured
+
+Start the consul agent in dev mode using the configuration file below for the
+'webserver' and 'ingress' services.
+
+```shell
+$ consul agent -dev -log-level=info -config-file=consul-services.json
+```
+
+#### consul-services.json
+```hcl
+cat > consul-services.json << EOF
+{
+  "services": [
+    {
+      "name": "ingress",
+      "port": 8080,
+      "connect": {}
+    },
+    {
+      "name": "webserver",
+      "connect": { "sidecar_service": {} },
+      "port": 8888
+    }
+  ]
+}
+EOF
+```
+This contains the service definitions for both the `webserver` and the
+`ingress` proxy (below). The names used are important as those are how you will
+refer to them later.
+
+## Start the Webserver (a Connect-unaware Service)
+
+First we need a web server to proxy to and for that we'll just use NGINX as we
+already have it installed to use as a proxy. We'll configure it to just serve
+up static files and be able to be run by a standard user.
+
+```shell
+$ /usr/sbin/nginx -p . -c nginx-server.conf
+```
+
+### Use a Sidecar to add it to the Connect mesh network
+
+Then start the sidecar for the `webserver` service.
+
+```shell
+$ consul connect proxy -sidecar-for webserver
+```
+
+**Note**: the argument to `-sidecar-for` needs to match the name registered
+with consul in the `consul-services.json` config file above.
+
+
+#### nginx-server.conf
+```nginx
```

Comment:
Are you including this example file in the repo or should the user create it?

Reply:
While working on the HAProxy doc I decided to use Python's included HTTP server instead of using NGINX as the web server. Doing so eliminates the need for this config file, simplifying things.

So instead of this config plus the command, it is replaced by...
`$ python -m SimpleHTTPServer`

It would still use the index.html, which I'd probably change to (like I'm using for HAproxy)... `Brought to you by NGINX!`

Reply:
I think I'm going to go ahead and make this change. I really don't see a downside.